R/shiny_fuse.R

```diff
@@ -0,0 +1,718 @@
+      values$annofuse_tbl <- annofuse_tbl
+
+      enhanced_annofuse_tbl <- annofuse_tbl
+      # enhancing the content of the table
+      enhanced_annofuse_tbl$Gene1A <- .multilink(enhanced_annofuse_tbl$Gene1A)
+      enhanced_annofuse_tbl$Gene1B <- .multilink(enhanced_annofuse_tbl$Gene1B)
+      values$enhanced_annofuse_tbl <- enhanced_annofuse_tbl
+    }
+
+    # Controls for plot panels -------------------------------------------------
+    output$plot_controls <- renderUI({
+      if (is.null(values$annofuse_tbl)) {
+        return(NULL)
+      } else {
+        tagList(
+          selectInput(
+            inputId = "af_cols",
+            label = "Grouping column",
+            choices = c("", colnames(values$annofuse_tbl)),
+            selectize = TRUE, multiple = FALSE, selected = "Fusion_Type"
+          ),
+          numericInput(
+            inputId = "af_n_topfusions",
+            label = "Nr top fusions",
+            value = 20, min = 1, max = 300, step = 1
+          ),
+          selectInput(
+            inputId = "af_countcol",
+            label = "Counting column",
+            choices = c("", colnames(values$annofuse_tbl)),
+            selectize = TRUE, multiple = FALSE, selected = "Sample"
+          ),
+        )
+      }
+    })
+
+    # Load annoFuse data file --------------------------------------------------
+    observeEvent(input$annofusedatasel, {
+      message("Reading in...")
+      values$annofuse_tbl <- read.delim(input$annofusedatasel$datapath)
+      values$enhanced_annofuse_tbl <- values$annofuse_tbl
+
+      # enhancing the content of the table
+      values$enhanced_annofuse_tbl$Gene1A <- .multilink(values$enhanced_annofuse_tbl$Gene1A)
+      values$enhanced_annofuse_tbl$Gene1B <- .multilink(values$enhanced_annofuse_tbl$Gene1B)
+
+      if (!is.null(values$data_pfam)) {
+        message("Creating domain information...")
+        values$ann_domain <- annoFuse::getPfamDomain(
+          standardFusioncalls = values$annofuse_tbl,
+          bioMartDataPfam = values$data_pfam, # must be pre-loaded
+          # partial overlapping domains are retained == "Partial" with keepPartialAnno=TRUE;
+          # if keepPartialAnno=FALSE then domain retained status == "No"
+          keepPartialAnno = TRUE
+        )
+      }
+    })
+
+    # Main interactive table for exploration -----------------------------------
+    output$table_annofuse <- DT::renderDataTable({
+      validate(
+        need(
+          !is.null(values$annofuse_tbl),
+          "Please upload the results of annoFuse to start the exploration"
+        )
+      )
+
+      DT::datatable(
+        values$enhanced_annofuse_tbl,
+        style = "bootstrap",
+        rownames = FALSE,
+        filter = "top",
+        selection = "single",
+        escape = FALSE,
+        options = list(
+          scrollX = TRUE,
+          pageLength = 25,
+          lengthMenu = c(5, 10, 25, 50, 100, nrow(values$enhanced_annofuse_tbl))
+        )
+      )
```

Comment:
Can we add a download option to this datatable with buttons=c("copy","csv","excel")? To just download the selected datatable rows. In case the user needs the flat file for the selected samples/genes ?

Reply:
Related to this, can we provide the PutativeDriverAnnoFuse_test_v14.tsv file as an example file? So that if ppl want to only look at PBTA samples that can be loaded and downloaded for the selected samples/genes .

Reply:
Re: downloadability/exportability -> yes, there are extensions for DT that handle that pretty elegantly.
I can look it up.

Re: example -> would be indeed a good idea, yes. Just wondering how to keep the size of the package compact & essential. But that would not hurt - the large size comes from the exons info

<img width="690" alt="image" src="https://user-images.githubusercontent.com/7447478/90429823-aedbac80-e0c6-11ea-84e6-2f9878e70001.png">

Reply:
Buttons are added in https://github.com/federicomarini/annoFuse/commit/a3beb88256c2cc0a0ea97a9e0f858b28ec5077d5